code/datums/callback_invoke.dm

```diff
@@ -0,0 +1,15 @@
+	//This is to reduce copypasta in callback.dm without proc overhead

```

Comment:
You're looking at something like:

```
#define INTERNAL_INVOKE \
if(!object) \
{ \
    return; \
} \
var/list/calling_arguments = arguments; \
if(length(args)) \
{ \
    if(length(arguments)) \
    { \
        calling_arguments = calling_arguments + args; \
    } \
    else \
    { \
        calling_arguments = args; \
    } \
} \
if(object == GLOBAL_PROC) \
{ \
    return call(delegate)(arglist(calling_arguments)); \
} \
return call(object, delegate)(arglist(calling_arguments)); \
```

You may need to indent this all by one, depends.